how-tos/add-extra-data-to-a-product.md

```diff
@@ -0,0 +1,106 @@
+ProductVariant.extend(schemaExtension);
+
+// Extend the CatalogProduct database schema, if your custom property will be on products
+CatalogProduct.extend(schemaExtension);
+
+// Extend the catalog variant database schemas, if your custom property will be on variants. There are two schemas for this one.
+VariantBaseSchema.extend(schemaExtension);
+CatalogVariantSchema.extend(schemaExtension);
+```
+
+### Extend GraphQL schemas
+
+- Extend the `Product` GraphQL type, if your custom property will be on products
+- Extend the `ProductVariant` GraphQL type, if your custom property will be on variants
+- Extend the `CatalogProduct` GraphQL type, if your custom property will be on products and is published to the catalog
+- Extend the `CatalogProductVariant` GraphQL type, if your custom property will be on variants and is published to the catalog
+- Create a GraphQL resolver for your property if it needs any transformation
+
+Refer to [How To: Extend GraphQL to add a field](./extend-graphql-to-add-field.md).
+
+### Register custom property as published
+
+*Skip this step if your property is not needed in the published catalog*
+
+A plugin can include a `catalog` object in `registerPlugin`, with `customPublishedProductFields` and `customPublishedProductVariantFields` that are set to arrays of property names. These will be appended to the core list of fields for which published status should be tracked. This is used to build the hashes that are used to display an indicator when changes need to be published.
+
+```js
+export default async function register(app) {
+  await app.registerPlugin({
+    catalog: {
+      customPublishedProductFields: ["myProperty"],
+      customPublishedProductVariantFields: ["myProperty"]
+    },
+    // other props
+  });
+}
+```
+
+### Register a function that publishes custom property
+
+*Skip this step if your property is not needed in the published catalog*
+
+```js
+import publishProductToCatalog from "./publishProductToCatalog";
+
+export default async function register(app) {
+  await app.registerPlugin({
+    functionsByType: {
+      publishProductToCatalog: [publishProductToCatalog]
+    },
+    // other props
+  });
+}
+```
+
+Where the `publishProductToCatalog` function looks something like this:
+
+```js
+export default function publishProductToCatalog(catalogProduct, { context, product, shop, variants }) {
+  catalogProduct.myProperty = product.myProperty;
+  // Also set on each catalogProduct.variants if necessary
+}
+```
+
+### Create a GraphQL mutation for setting your custom property value
+
+Refer to [How To: Create a new GraphQL mutation](./create-graphql-mutation.md). After mutating the `Product` document in your function, you must also call `context.mutations.hashProduct(productId, context.collections, false)`. This will update the current product hash, causing the operator UI to indicate that there are changes needing publishing.
```

Comment:
Just to follow specifications: `context.mutations.hashProduct(<productId>, context.collections, false)`

Reply:
Updated with change